Partner Portal: Update page layout to handle wider screens.

jkguidaven opened the pull request:
Related to 1202619025189113-as-1205116564628275

The current Partner portal layout does not work well on a wider screen. This PR addresses this issue by refactoring the current layout component so it can scale with wider screens.

**Before**
<img width="1511" alt="Screen Shot 2023-08-11 at 7 04 53 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/af362315-8102-4ab3-a155-7334ad4d239e">
**After**
<img width="1522" alt="Screen Shot 2023-08-14 at 8 36 50 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/33f8b152-5366-4606-8162-9454038afbfc">

## Proposed Changes

* Add Layout **Top** and **Body** components to wrap the contents with proper margins/paddings and border.
* Update existing consuming components to use the additional layout components.


## Testing Instructions

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

* Use the Jetpack cloud live link below and test the following screens. Make sure you zoom-out your browser to simulate wider screens.

| Screen  | URL |
| ------------- | ------------- |
| Billing  | /partner-portal/billing  |
| Issue License  | /partner-portal/issue-license  |
| Assign License  | /partner-portal/assign-license  |
| Payment Methods | /partner-portal/payment-methods |
| Invoices | /partner-portal/invoices |
| Prices | /partner-portal/prices |

* Also make sure the Mobile view is not affected with this changes and works exactly the same.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1205254810113133

jkguidaven committed:
Update layout so it spans 100% except the content.

jkguidaven committed:
Fix missing background color.

jkguidaven committed:
Fix missing layout update on company details page.

TheSteveK reviewed (changes requested):
I like the approach of pushing this out to individual. template files to keep things a little cleaner.  Non-blocking, have you thought about also breaking out the CSS for the templates now that there are multiple files?

Pages like the licenses page work as advertised, but others like the pricing page, I don't see a difference between the two as far as width.  But, I did notice there's a discrepency in horizontal spacing as demonstrated in t

https://github.com/Automattic/wp-calypso/assets/12895386/eddbb1f5-6453-407b-8635-c8a7fa20f915

he video below:

jkguidaven committed:
Fix descrepancies with horizontal spacing.

jkguidaven commented:
Thanks for reviewing my PR @TheSteveK.

> I like the approach of pushing this out to individuals. Template files to keep things a little cleaner. Non-blocking, have you thought about also breaking out the CSS for the templates now that there are multiple files?

Good question. I initially considered breaking it to separate CSS files to match individual component files but decided not to since the overall CSS size is relatively small, and I did not want to introduce more files on this PR which I already did with the template files. I'll leave it as is for now and decide later if we need to break it if it grows.

> Pages like the licenses page work as advertised, but others like the pricing page, I don't see a difference between the two as far as width. But, I did notice there's a discrepency in horizontal spacing as demonstrated in t

Good catch, this has been fixed.

TheSteveK commented:
> Good question. I initially considered breaking it to separate CSS files to match individual component files but decided not to since the overall CSS size is relatively small, and I did not want to introduce more files on this PR which I already did with the template files. I'll leave it as is for now and decide later if we need to break it if it grows.

That's a good point, it did take a while to dig through those files :)

TheSteveK reviewed (approved):
Thanks for fixing!  🚢

jkguidaven committed:
Fix issue w/ license page padding.

jkguidaven committed:
Fix payment method add screen card margin.

jkguidaven commented:
Hi @TheSteveK thanks for the review. I added some additional fixes. Can you have another look? 🙏

yashwin reviewed (commented):
This is looking good.

However, I found a couple of issues.

1) The banner is aligned with the below content

<img width="791" alt="Screenshot 2023-09-01 at 8 16 10 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/75feac5c-da87-4698-9059-9780cda4dd4a">

2) This screen is overflowing. 

<img width="375" alt="Screenshot 2023-09-01 at 8 23 08 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/5e205e86-c619-4464-babe-dcab83ce09a6">

While this was overflowing before this change, it was visible.

<img width="368" alt="Screenshot 2023-09-01 at 8 23 19 AM" src="https://github.com/Automattic/wp-calypso/assets/10586875/a5eb1454-9f6a-41b4-86e0-cd1ba5f11355">

I also noticed some CSS styles not using RTL-friendly attributes, which I think is okay since it is not a part of this PR.

jkguidaven committed:
Fix banner not aligning with top header.

jkguidaven committed:
Update CSS so it is RTL friendly.

jkguidaven commented:
@yashwin, thanks for the review 🙏 . I fixed the issue with/ the banner and improved the CSS to be RTL-friendly. While I couldn't reproduce the problem with/ overflowing, I think it was fixed when I rebased it, but can you have another look?

yashwin commented:
@jkguidaven: Thanks for making the changes and fixing the issue!

> While I couldn't reproduce the problem with/ overflowing, I think it was fixed when I rebased it, but can you have another look?

I can still see this issue. In fact, I see the padding has increased on all these screens. I think we can keep the same padding as before, especially on the small screens.


<table>
<tr>
<th>
Before
</th>
<th>
After
</th>
</tr>


<tr>
<td>
<img width="319" alt="Screenshot 2023-09-12 at 2 32 52 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/08187e5a-b58e-4000-8d55-bcacf092558f">
</td>
<td>
<img width="319" alt="Screenshot 2023-09-12 at 2 32 40 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/5ba314a1-0ba7-4f2d-b4e9-e6ca5c3addc9">
</td>
</tr>

<tr>
<td>
<img width="319" alt="Screenshot 2023-09-12 at 2 34 08 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/d915ab47-abe8-43a8-afc3-7b2255c38954">
</td>
<td>
<img width="319" alt="Screenshot 2023-09-12 at 2 33 59 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/b9192f5d-0eb1-40d1-baea-0eecae8f1d3f">
</td>
</tr>

<tr>
<td>
<img width="543" alt="Screenshot 2023-09-12 at 2 37 37 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/838614c4-d953-4a95-9db2-1bcf8d403056">
</td>
<td>
<img width="543" alt="Screenshot 2023-09-12 at 2 37 32 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/36ca7a58-6bcc-4718-a797-7badef39577c">
</td>
</tr>


</table>

yashwin commented:
@jkguidaven: Please discard my above comment. I think the Jetpack Cloud link was not updated. I tried locally and it looks good now.

jkguidaven committed:
Fix alignment of mobile header on some of the pages.

yashwin reviewed (approved):
LGTM 🚀 Thanks for fixing this 👍
